### PR TITLE
Implement programming question models

### PR DESCRIPTION
**Pull request by lowjoel**
Something so simple, yet so much to write...

### CODE REVIEW

**Comment by lowjoel** [comment]
@allenwq this should be good.

File: db/schema.rb [hunk 1]
```diff
@@ -196,10 +198,50 @@
...
+  end
+
+  create_table "course_assessment_answer_programming_auto_grading_test_results", force: :cascade do |t|
+    t.integer "auto_grading_id", null: false, index: {name: "fk__course_assessment_answe_57b22f114b54911fd4e1519680ebfd49"}, foreign_key: {references: "course_assessment_answer_programming_auto_gradings", name: "fk_course_assessment_answer_e3d785447112439bb306849be8690102", on_update: :no_action, on_delete: :no_action}
+    t.integer "test_case_id",    index: {name: "fk__course_assessment_answe_29a2568d5e2fb3a47c0561815786f9ab"}, foreign_key: {references: "course_assessment_question_programming_test_cases", name: "fk_course_assessment_answer_bbb492885b1e3dca4433b8af8cb95906", on_update: :no_action, on_delete: :no_action}
+    t.boolean "passed",          null: false
+    t.text    "message"
+  end
+
+  create_table "course_assessment_answer_programming_files", force: :cascade do |t|
```
Comment (allenwq) [inline db/schema.rb]: So assessments will use its own file model ?  Will this do anything with the attachment in future ?

Comment (lowjoel) [inline db/schema.rb]: No. these are actual contents of files. attachments will be separate.

Comment (allenwq) [inline db/schema.rb]: Why we need the actual content of file ? how about binary files ?

Comment (lowjoel) [inline db/schema.rb]: these are programming questions, students should be writing their solution directly inside Coursemology. Eventually I'll allow students to upload zips and we keep the text files.

Comment (allenwq) [inline db/schema.rb]: I see, thanks !

**Pull request merged**